diff --git a/net/ipv4/netfilter/Kconfig b/net/ipv4/netfilter/Kconfig
index f17b40211..99f691a67 100644
--- a/net/ipv4/netfilter/Kconfig
+++ b/net/ipv4/netfilter/Kconfig
@@ -239,6 +239,15 @@ config IP_NF_TARGET_NETMAP
 	(e.g. when running oldconfig). It selects
 	CONFIG_NETFILTER_XT_TARGET_NETMAP.

+config IP_NF_TARGET_FULLCONENAT
+	tristate "FULLCONENAT target support"
+	depends on NETFILTER_ADVANCED
+	select NETFILTER_XT_TARGET_FULLCONENAT
+	---help---
+	This is a backwards-compat option for the user's convenience
+	(e.g. when running oldconfig). It selects
+	CONFIG_NETFILTER_XT_TARGET_FULLCONENAT.
+
 config IP_NF_TARGET_REDIRECT
 	tristate "REDIRECT target support"
 	depends on NETFILTER_ADVANCED
diff --git a/net/netfilter/Kconfig b/net/netfilter/Kconfig
index 91efae88e..17f5c748a 100644
--- a/net/netfilter/Kconfig
+++ b/net/netfilter/Kconfig
@@ -956,6 +956,14 @@ config NETFILTER_XT_TARGET_NETMAP

 	To compile it as a module, choose M here. If unsure, say N.

+config NETFILTER_XT_TARGET_FULLCONENAT
+	tristate '"FULLCONENAT" target support'
+	depends on NF_NAT
+	---help---
+	Full Cone NAT
+
+	To compile it as a module, choose M here. If unsure, say N.
+
 config NETFILTER_XT_TARGET_NFLOG
 	tristate '"NFLOG" target support'
 	default m if NETFILTER_ADVANCED=n
diff --git a/net/netfilter/Makefile b/net/netfilter/Makefile
index 4fc075b61..2b588d5a5 100644
--- a/net/netfilter/Makefile
+++ b/net/netfilter/Makefile
@@ -209,3 +209,6 @@ obj-$(CONFIG_IP_SET) += ipset/

 # IPVS
 obj-$(CONFIG_IP_VS) += ipvs/
+
+# Full cone NAT
+obj-$(CONFIG_NETFILTER_XT_TARGET_FULLCONENAT) += xt_FULLCONENAT.o
